### DISCUSSION TITLE
Challenges with Installing RustDesk Relay Server Using Subdomain and Seeking Secure Connection for Web Console

### DISCUSSION THREAD

**Discussion by isential-gmbh** [rustdesk/rustdesk-server-pro#227] [General]
I'm using a VPS from Strato for the RustDesk Relay Server. It's running Ubuntu 22.04.

I can only install the RustDesk Relay Server if I use the server's IP address. But I'd like to use a subdomain. However, I get an error during installation when I try to use a subdomain. No matter what I do, I keep getting the error message:

"Sorry, the TLS certificate for rustdesk.example.tld failed to generate! Please check that port 80/443 are correctly port forwarded, and that the DNS record points to this server's IP."

Here I have already described the problem: https://github.com/rustdesk/rustdesk-server-pro/discussions/220

As mentioned before, using the public IP address works.

Now, I can access the web console with "http://IP-Address:21114". It works, but I don't like that it's using "http" instead of "https". That's why I wanted to use the subdomain "rustdesk.example.tld" because I understand that https should be possible with it. Is that correct?

Unfortunately, I'm not a Linux expert, and I'm even less familiar with configuring nginx as a reverse proxy. However, I imagine I could use nginx to handle https up to the RustDesk Relay Server, internally forwarding the traffic over http. Is that possible, and if so, how?

Or are there other ways to operate the web console over a secure connection?

**Comment by rustdesk** [comment]
I will make a detailed tutorial for this, please allow me one more day.

**Reply by rustdesk** [reply]
https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/faq/#set-up-https-for-web-console-manually

**Comment by ZetaWaves** [comment]
Use NGINX proxy to proxy the http request over https instead. The simple install script does this automatically.
